doc/governance.rst

```diff
@@ -0,0 +1,130 @@
+
+.. _governance:
+
+===========================================
+Scikit-learn governance and decision-making
+===========================================
+
+The purpose of this document is to formalize the governance process used by the
+scikit-learn project, to clarify how decisions are made and how the various
+elements of our community interact.
+This document establishes a decision-making structure that takes into account
+feedback from all members of the community and strives to find consensus, while
+avoiding any deadlocks.
+
+This is a meritocratic, consensus-based community project. Anyone with an
+interest in the project can join the community, contribute to the project
+design and participate in the decision making process. This document describes
+how that participation takes place and how to set about earning merit within
+the project community.
+
+Roles And Responsibilities
+==========================
+
+Contributors
+------------
+Contributors are community members who contribute in concrete ways to the
+project. Anyone can become a contributor, and contributions can take many forms
+– not only code – as detailed in the `contributors guide <contributing>`_.
+
+Core developers
+---------------
+Core developers are community members who have shown that they are dedicated to
+the continued development of the project through ongoing engagement with the
+community. Being a core developer  allows contributors to more easily carry on
+with their project related activities by giving them direct access to the
+project’s repository and is represented as being an organization member on the
+scikit-learn `GitHub organization <https://github.com/orgs/scikit-learn/people>`_.
+Core developers are expected to review code
+contributions, can merge approved pull requests, can cast votes for and against
+merging a pull-request, and can be involved in deciding major changes to the
+API.
+
+New core developers can be nominated by any existing core developers. Once they
+have been nominated, there will be a vote by the current core developers.
+Voting on new core developers is one of the few activities that takes place on
+the project's private management list. While it is expected that most votes
+will be unanimous, a two-thirds majority of the cast votes is enough. The vote
+needs to be open for at least 1 week.
+
+Core developers that have not contributed to the project (commits or GitHub
```

Comment:
As a side note, this is a bit in contradiction with "not only code contributions are valued." This doesn't include any organization of sprints, fundraising, etc.

Reply:
I think we can change that to "having actively contributed (commits, comments or organizational and non-code contributions)". I'm slightly hesitant about changing things now, but I think it would be consistent with the spirit of the document to rephrase this.

Reply:
Recall, though, that the emeritus status is only *negotiated* ("will be asked if they want to become emeritus"), triggered by this criterion. Explicitly using a criterion of commits or comments makes it much easier to measure activity by which this emeritus proposition is triggered. A vague criterion is not especially helpful here. There is nowhere that says the emeritus status will be forced, though presumably this would follow the same as any other personnel dispute (not explicitly covered in this document), such as removing core dev rights.

Reply:
I think the suggestion was more in terms of what he document communicates. I'm not sure we'll use a very precise criterion anyway (unless you wanted to write a bot ;). I figured it would be more that either we'll do a (yearly?) spring cleaning or someone thinks "has been awhile since this person was around"?
I don't have strong feelings either way, though.